server/impl/src/main/java/com/walmartlabs/concord/server/process/queue/ProcessQueueDao.java

```diff
@@ -278,6 +278,31 @@ public boolean updateStatus(DSLContext tx, ProcessKey processKey, ProcessStatus
         return i == 1;
     }
 
+    public boolean updateStatus(List<ProcessKey> processKeys, List<ProcessStatus> expected, ProcessStatus status) {
+        return txResult(tx -> {
+            List<UUID> instanceIds = processKeys.stream()
+                    .map(PartialProcessKey::getInstanceId)
+                    .collect(Collectors.toList());
+
+            UpdateConditionStep<ProcessQueueRecord> q = tx.update(PROCESS_QUEUE)
+                    .set(PROCESS_QUEUE.CURRENT_STATUS, status.toString())
+                    .set(PROCESS_QUEUE.LAST_UPDATED_AT, currentOffsetDateTime())
+                    .set(PROCESS_QUEUE.LAST_RUN_AT, createRunningAtValue(status))
+                    .where(PROCESS_QUEUE.INSTANCE_ID.in(instanceIds));
+
+            if (expected != null) {
```

Comment:
Should it check if the list is empty as well? `expected != null && !expected.isEmpy()`?

Reply:
it's old code :) I think jooq check empty lists in `in` clause. but added empty list check.